### DISCUSSION TITLE
2022-05-04 Meeting Agenda, Recording & Notes

### DISCUSSION THREAD

**Discussion by andy-blum** [ddev/ddev#3651] [DDEV Advisory Group]
Previous meeting: #3650 
Next meeting: #3800

## Agenda:
- Turn on recording, ask for a minutes-taker
- Set next meeting date (or recurring)
- v1.19.1 and upcoming v1.19.2 situation
- Upcoming Colima/Lima upgrades (9p mount)
- Slower April because I was gone
- It&#39;s time for an official announcement of Platform.sh&#39;s sponsorship
- ddev add-ons status, official and unofficial
- Upstream donations 10% status at this point
- Financial status
- Questions:
  - More nominations for upstream projects to donate to with 10% of GitHub Sponsors
- Anything on anybody&#39;s mind!

## Minutes:

- Recording started
- No need to set new meeting date (bi-monthly cadence going forward)
- If using Colima, there&#39;s a big colima update coming soon
  - We&#39;ll be able to have a real mount type instead of sshfs
  - Biggest change is updated networking backend
  - Andrew Berry: Been testing, works with mutagen
  - Doesn&#39;t work with symlinks
  - Randy: most issues with sshfs don&#39;t happen for users, but pop up in testing since things move so quickly. Thanks to Andrew for being on top of that since the very beginning
- Randy was gone all April biking the Camino de Santiago
  - Toyed with the idea of taking a computer to continue helping in support forums
  - Decided against that and **took a much deserved break [editor&#39;s note]**
  - At least 5 different things rotted in his absence that needed addressed that needed fixing for tests
  - Cary: what you&#39;re saying points to the need to have another developer on the project. If you won the lottery and retired, we&#39;d need someone who could carry the project. This would also allow you to just leave your devices at home next time and not have to worry about the project at all
  - Randy: we don&#39;t know the lifetime of the project or of Randy. As the project grows, we need a more sustainable and robust situation. It&#39;s not just being able to do the work, but also understanding the scope of the project. For example, I&#39;m not willing to take on a GUI because I don&#39;t want to expand the scope of the project, but someone else could. It would take the community a lot of time to be able to fix some of these cascading issues/intricacies.
  - Cary: I don&#39;t think we should expand the scope in any way until we&#39;ve been able to bring another person/people up to speed on the project
  - Randy: Hard to find people with extra cycles to chip in, we have the potential to pay people with the localdev foundation or via platform.sh
- Time for official announcement of Platform.sh sponsorship
  - It&#39;s only fair for the community to know about all the funding changes since the closure of DRUD
  - We&#39;ll get Platform&#39;s PR people on it, share within this group, then share publicly
- DDEV add-ons
  - ddev get --list
    - Lists all addons, with descriptions
    - Currently 12 add-ons listed
    - 9 add-ons marked as &quot;official, maintained&quot;
  - Joao: At 1x we&#39;re not always trying to get the latest ddev, but now try to stay up-to-date for the recipes/add-ons
  - Randy: The strategy isn&#39;t perfect, we don&#39;t have update capabilities, we don&#39;t know if you&#39;re overwriting existing configs. But they are maintained and tested (at least the official addons). A process, not a feature change.
  - Joao: Lando has had this feature, which was a missing capability in ddev.
  - R: However, lando doesn&#39;t test, it&#39;s really a community contribution.
- Upstream donations and contributions
  - [https://github.com/drud/ddev/discussions/3682](https://github.com/drud/ddev/discussions/3682)
  - A number are not possible to donate to
  - The current list: [https://github.com/drud/ddev/discussions/3682#discussioncomment-2687327](https://github.com/drud/ddev/discussions/3682#discussioncomment-2687327)
  - Goal: $160 a month given what we want to accomplish.
  - ![](RackMultipart20220504-1-4un1s4_html_7d1f80d9024ac9d8.png)
  - R: Thinks we could go higher than this. We need a clear goal and a clear way to accomplish that goal. Platform may be willing to participate in that.
- Accounting update:
  - ~$7100 in the bank.
  - The income statement is a bit messy, but shows our direct public support.
  - Turned on macstadium instance while gone. $130 / month. I like it, but it&#39;s supposed to be for free, we&#39;ve applied but not happened yet.
  - We had gone to the free CircleCI plan, but turned it back on as we ran into some limitations.
  - Carey: Where does this put us in terms of test runners?
    - R: Two developments. I spent some time looking for an internet-based Windows solution. Didn&#39;t find a good solution. Should be making a plan to replace some of the Windows test runners with Windows PCs. ~$400 each. On macOS, you can now finally install Docker from the command line… but we can&#39;t do that from GitHub. With their test runners, you end up with a backlog of test runners.
    - We will likely get a free MacStadium + one we pay for.
    - R: Will put this on my list of tasks.
    - A: Do we need a Windows expert to help with setup?
      - R: I&#39;ve tried a lot of many different solutions. The real problem is licensing, because you need nested virtualization which typically isn&#39;t enabled for Docker Desktop / hyperv.
      - The only one I sort of got to work was way to slow.
    - A: What about colocation?
      - R: The current systems are already accessible. I use a remote desktop, so if I have a partner managing the test bots. I restart and fix them periodically, was amazed we made it through the month!
      - We haven&#39;t scaled to the point where colo benefits us significantly.
    - R: We spend a lot of time on traditional (not WSL2) Windows support, it would be good to only support WSL2. Most of the test resources are not in WSL2.
    - Carey: What if we dropped non-WSL2 support?
      - R: Disrupts workflows, likely upsetting people.
      - Issues are not knowing linux, WSL2, or corporate restrictions on WSL2.
    - C: How do we reach out to ddev users for feedback?
      - Drupal has succeeded on the strength of its community. DDEV is the same, but much earlier and smaller.
      - Getting more people, but at a lower level of commitment. How can we do that?
      - R: The point of this group is to get some of that feedback, people who are leading or working with teams to feed information back into here, discord, github.
      - Joao: We mention a new version of ddev when running. We could mention a new poll in the same place.
      - R: Great idea! Let&#39;s file an issue for this.
      - R: When there are issues, if they don&#39;t open an issue we don&#39;t know what problem they failed to solve.
      - We get a lot of random issues opened that are very interesting to learn from. Would appreciate anyone who is willing to subscribe to all issues in the repository.
      - Ofer: A way to bring people together to ddev is through drupalpod. A way to make it easier to set up. We could have some communication there. Make users more aware of the ddev part of the infrastructure.
        - R: Seems like in the drupalpod chat people are sometimes confused over what it is, how ddev is included. Shows that drupalpod is succeeding.
        - O: What about returning customers, who may be interested in more advanced usage. Show them how easy ddev makes the process.
      - R: Would appreciate continued feedback on how the team can do that.
      - O: I like automatic telemetry, but the public does not.
      - R: We use Segment and Amplify, if you opt in. Some good information. Segment costs though, $150 / month at least. We&#39;re on the free version now. We know projects but not commands, only via ddev start and not other commands.
      - A: Home Assistant analytics has done this within an OSS community, and they&#39;ve had lots of opt-in to it. They make all their analytics public.
- Anything else?
  - Would love for people to poke through [https://github.com/drud/ddev/milestone/54](https://github.com/drud/ddev/milestone/54) - they are mostly in priority order. Or, if there are issues that are being neglected.
  - Carey: Replacing the router with traefik.
  - R: Its in the milestone! I&#39;m studying it. If friction can be avoided, it&#39;s doable, so it could possibly be an option in 1.20 and then default in 1.21.
    - Default way to use it requires mounting the docker socket into the container. Not great from a security standpoint, what we do in the current router, but would like to remove that.
    - [Stop mounting docker.sock into ddev-router (security)](https://github.com/drud/ddev/issues/2653)
    - We&#39;ll have to figure out HTTPs to not break users.
  - Trules: Have we landed on something regarding nvm?
    - Ddev 1.19 has full nvm support!
    - 1.19.2 has node 18 as the system node version
  - Christian: It could be possible to run without the docker socket. It&#39;s used to detect the labels on the containers. But then we have to replace it with some other technology like consul or kv storage.
    - R: Thinks we know enough from ddev on the host side that it could configure and restart the router as needed.
  - Ofer: node 18 as the default?
    - R: Yes, in 1.20? Whenever 18 goes LTS.
    - Mariadb is going to have to go to 10.4 as the default soon.
    - Also, going to PHP 8.0, but it depends when we release for all of these.
    - Would love input on when 17 EOL hits (June 1st). Less than 1 month.
    - Don&#39;t want builds to break when they remove the repositories.
  - O: Can look at the maintainer of search API solr. There&#39;s a ddev contrib pull request.
    - R: Been working on a long time: [https://github.com/drud/ddev-contrib/pull/195](https://github.com/drud/ddev-contrib/pull/195)
    - The config changes didn&#39;t work with arm64, so this PR makes it work with both.
    - More complex due to adding zookeeper. Assumes 2+ solr servers.
  - Robert: Can recipes add modules to the base setup?
    - Yes, they can.

# Recording

[Recording available here](https://www.dropbox.com/s/nmmaxu5dn04hche/GMT20220504-145056_Recording_1760x900.mp4?dl=0)

**Comment by tyler36** [comment]
~~Is "2022-05-04 Meeting Agenda, Recording & Notes" the right title?~~

Sorry, I see it is preparation for next meeting.

**Reply by rfay** [reply]
Andy is SO on top of things!

**Comment by rfay** [comment]
Thanks @andy-blum @deviantintegral !

**Comment by deviantintegral** [comment]
Here is the link to the recording, @andy-blum can you edit it into the post?

https://www.dropbox.com/s/nmmaxu5dn04hche/GMT20220504-145056_Recording_1760x900.mp4?dl=0
